server/asset/eth/common.go

```diff
@@ -78,15 +78,17 @@ func (ss SwapState) String() string {
 
 // DecodeCoinID decodes the coin ID into flags, a contract address which may be
 // zeroed, and hash that represents either a secret or txid depending on flags.
-func DecodeCoinID(coinID []byte) (CoinIDFlag, common.Address, []byte, error) {
+func DecodeCoinID(coinID []byte) (CoinIDFlag, *common.Address, []byte, error) {
+	var addr common.Address
 	if len(coinID) != coinIDSize {
-		return 0, common.Address{}, nil, fmt.Errorf("coin ID wrong length. expected %d, got %d",
+		return 0, &addr, nil, fmt.Errorf("coin ID wrong length. expected %d, got %d",
```

Comment:
Looks like we can/should be returning nil for the `*common.Address` in the event of non-nil error return, but I haven't really gotten into this yet.

Reply:
Returning nil.